grants_tagger/scispacy_meshtagger.py

```diff
@@ -0,0 +1,59 @@
+"""
+Wrapper for scispacy mesh tagger. SciSpacy implements mesh
+tagging as a named entity recognition and linking step so
+the class also transforms the problem to multi label classification
+"""
+from collections import defaultdict
+
+from scispacy.linking import EntityLinker
+from scispacy.abbreviation import AbbreviationDetector
+from sklearn.metrics import f1_score
+from scipy.sparse import csr_matrix
+import scispacy
+import spacy
+
+
+class SciSpacyMeshTagger():
+    def __init__(self, mesh_tags, threshold=2):
+        """
+        mesh_tags: list of mesh UI descriptros e.g. [D00001]
+        """
+        self.mesh_tags = mesh_tags
+        self.threshold = threshold
+
+    def fit(self, *_):
+        nlp = spacy.load("en_core_sci_sm")
+        abbreviation_resolver = AbbreviationDetector(nlp)
+        nlp.add_pipe(abbreviation_resolver)
+        linker = EntityLinker(resolve_abbreviations=True, name="mesh")
+        nlp.add_pipe(linker)
+        self.nlp = nlp
+
+    def _get_tags(self, doc):
+        mesh_ents = defaultdict(int)
```

Comment:
is `mesh_ent_prob` an int then? I assumed it was a probability from the name

Reply:
hmm, this sounds like a bug, well spotted, let me double check